integration_tests/src/main/python/conftest.py

```diff
@@ -307,6 +299,30 @@ def pytest_configure(config):
 print(f"Starting with OOM injection seed: {oom_random_injection_seed}. "
       "Set env variable SPARK_RAPIDS_TEST_INJECT_OOM_SEED to override.")
 
+# Returns a tuple (seed, permanent) with the seed that test `item` should use given a 
+# possibly defined `datagen_overrides`, and if the seed choice is due to an override, 
+# whether that override is marked as `permanent`
+def get_effective_seed(item, datagen_overrides):
+    if datagen_overrides:
+        permanent_override = False 
+        # if the override is marked as permanent (cannot be disabled) or the marker is temporary
+        # and we haven't disabled markers
+        try:
+            permanent_override = datagen_overrides.kwargs["permanent"]
+        except KeyError:
+            pass
+
+        if permanent_override or not item.config.getoption("datagen_seed_override_disabled"):
+            try:
+                seed = datagen_overrides.kwargs["seed"]
+            except KeyError:
+                raise Exception("datagen_overrides requires an override seed value")
+
+            override_seed = datagen_overrides.kwargs.get('condition', True)
+            if override_seed:
+                return (seed, permanent_override)
+    return (_test_datagen_random_seed_init, False)
+
 def pytest_collection_modifyitems(config, items):
     r = random.Random(oom_random_injection_seed)
     for item in items:
```

Comment:
Could inspect `items` here if all nodeid refer to the same function https://github.com/NVIDIA/spark-rapids/issues/10089#issuecomment-1871397360, and if it's only one function. 

If there is only one function and the DATAGEN_SEED was supplied externally, then the scenario is that the user is executing a repro step from a bug like https://github.com/NVIDIA/spark-rapids/issues/9992 and the DATAGEN_SEED should be the seed unconditionally to make the bug reproduce without any switches

Reply:
Let me know if my reply to that comment makes sense https://github.com/NVIDIA/spark-rapids/issues/10089#issuecomment-1871403908. I didn't want to have an environment variable that has different meanings, instead I'd like to make it explicit. Yes that's something else we have to learn, but I think this way at least the approach is always consistent.